From bd320170fe05acdc61560617b2c86ea46806aa33 Mon Sep 17 00:00:00 2001
From: Mikhail Efremov <sem@altlinux.org>
Date: Wed, 29 Aug 2012 21:29:50 +0400
Subject: [PATCH] Don't control disk's spin speed.

Avoid dependence on Udisks due it is obsoleted by Udisks2, but
Udisks2 don't have such option.
Install both Udisks and Udisks2 seems bad idea.
---
 xfce4-power-manager/settings/xfpm-settings-main.c |    4 ++--
 xfce4-power-manager/src/Makefile.am               |    2 --
 xfce4-power-manager/src/xfpm-manager.c            |   10 ++++++++++
 3 files changed, 12 insertions(+), 4 deletions(-)

diff --git a/xfce4-power-manager/settings/xfpm-settings-main.c b/xfce4-power-manager/settings/xfpm-settings-main.c
index 995be00..c7a6f20 100644
--- a/xfce4-power-manager/settings/xfpm-settings-main.c
+++ b/xfce4-power-manager/settings/xfpm-settings-main.c
@@ -161,8 +161,8 @@ int main (int argc, char **argv)
 	    has_power_button = xfpm_string_to_bool (g_hash_table_lookup (config_hash, "power-button"));
 	    has_hibernate_button = xfpm_string_to_bool (g_hash_table_lookup (config_hash, "hibernate-button"));
 	    can_shutdown = xfpm_string_to_bool (g_hash_table_lookup (config_hash, "can-shutdown"));
-	    can_spin_down = xfpm_string_to_bool (g_hash_table_lookup (config_hash, "can-spin"));
-	    devkit_disk = xfpm_string_to_bool (g_hash_table_lookup (config_hash, "devkit-disk"));
+	    can_spin_down = FALSE;
+	    devkit_disk = FALSE;
 	    
 	    g_hash_table_destroy (config_hash);
 	    
diff --git a/xfce4-power-manager/src/Makefile.am b/xfce4-power-manager/src/Makefile.am
index eaca181..aa993d8 100644
--- a/xfce4-power-manager/src/Makefile.am
+++ b/xfce4-power-manager/src/Makefile.am
@@ -16,8 +16,6 @@ xfce4_power_manager_SOURCES =                   \
 	xfpm-battery.h				\
 	xfpm-xfconf.c				\
 	xfpm-xfconf.h				\
-	xfpm-disks.c				\
-	xfpm-disks.h				\
 	xfpm-console-kit.c			\
 	xfpm-console-kit.h			\
 	egg-idletime.c				\
diff --git a/xfce4-power-manager/src/xfpm-manager.c b/xfce4-power-manager/src/xfpm-manager.c
index 472ee3e..28051cb 100644
--- a/xfce4-power-manager/src/xfpm-manager.c
+++ b/xfce4-power-manager/src/xfpm-manager.c
@@ -40,7 +40,9 @@
 
 #include "xfpm-power.h"
 #include "xfpm-dbus.h"
+/*
 #include "xfpm-disks.h"
+*/
 #include "xfpm-dpms.h"
 #include "xfpm-manager.h"
 #include "xfpm-console-kit.h"
@@ -82,7 +84,9 @@ struct XfpmManagerPrivate
     XfpmBacklight   *backlight;
     XfpmConsoleKit  *console;
     XfpmDBusMonitor *monitor;
+	/*
     XfpmDisks       *disks;
+	*/
     XfpmInhibit     *inhibit;
     EggIdletime     *idle;
 #ifdef HAVE_DPMS
@@ -133,7 +137,9 @@ xfpm_manager_finalize (GObject *object)
     g_object_unref (manager->priv->client);
     g_object_unref (manager->priv->console);
     g_object_unref (manager->priv->monitor);
+	/*
     g_object_unref (manager->priv->disks);
+	*/
     g_object_unref (manager->priv->inhibit);
     g_object_unref (manager->priv->idle);
     
@@ -523,7 +529,9 @@ void xfpm_manager_start (XfpmManager *manager)
     manager->priv->conf = xfpm_xfconf_new ();
     manager->priv->console = xfpm_console_kit_new ();
     manager->priv->monitor = xfpm_dbus_monitor_new ();
+	/*
     manager->priv->disks = xfpm_disks_new ();
+	*/
     manager->priv->inhibit = xfpm_inhibit_new ();
     manager->priv->idle = egg_idletime_new ();
     
@@ -615,8 +623,10 @@ GHashTable *xfpm_manager_get_config (XfpmManager *manager)
 		  "has-lid", &has_lid,
 		  NULL);
 
+	/*
     can_spin = xfpm_disks_get_can_spin (manager->priv->disks);
     devkit_disk = xfpm_disks_kit_is_running (manager->priv->disks);
+	*/
     
     has_battery = xfpm_power_has_battery (manager->priv->power);
     has_lcd_brightness = xfpm_backlight_has_hw (manager->priv->backlight);
-- 
1.7.3.3

